include/range/v3/numeric/partial_sum.hpp

```diff
@@ -30,16 +30,14 @@ namespace ranges
     inline namespace v3
     {
         template<typename I, typename O, typename BOp = plus, typename P = ident,
-            typename V = value_type_t<I>,
-            typename X = concepts::Invocable::result_t<P&, V>,
-            typename Y = concepts::Invocable::result_t<BOp&, X, X>>
+            typename X = projected<projected<I, coerce<value_type_t<I>>>, P>,
+            typename Y = reference_t<X>>
         using PartialSummable = meta::strict_and<
+            SemiRegular<detail::decay_t<Y>>,
             InputIterator<I>,
-            OutputIterator<O, X>,
-            Invocable<P&, V>,
-            CopyConstructible<uncvref_t<X>>,
-            Invocable<BOp&, X, X>,
-            Assignable<uncvref_t<X>&, Y>>;
+            OutputIterator<O, detail::decay_t<Y>&>,
```

Comment:
`OutputIterator<O, T const&>` activates the "implicit expression variants" wording and therefore requires `*out = acc;` to also be valid and not modify `acc`. (Irony: if writing through the output iterator might modify the accumulator, then an implementation must output a copy of the accumulator which would roughly require `SemiRegular` ;)